src/test/java/com/razorpay/CustomerClientTest.java

```diff
@@ -295,4 +296,168 @@ public void testDeleteToken() throws IOException, RazorpayException {
+                "}");
+
+        String mockedResponseJson = "{\n" +
+                "  \"entity\": \"customer\",\n" +
+                "  \"amount\": \"500000\",\n" +
+                "  \"customer\": {\n" +
+                "    \"id\": \"KkBhM9EC1Y0HTm\",\n" +
+                "    \"contact\": \"+918220722114\"\n" +
+                "  },\n" +
+                "  \"instruments\": [\n" +
+                "    {\n" +
+                "      \"method\": \"emi\",\n" +
+                "      \"issuer\": \"HDFC\",\n" +
+                "      \"type\": \"debit\",\n" +
+                "      \"eligibility_req_id\": \"elig_KkCNLzlNeMYQyZ\",\n" +
+                "      \"eligibility\": {\n" +
+                "        \"status\": \"eligible\"\n" +
+                "      }\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+        try {
+            mockResponseFromExternalClient(mockedResponseJson);
+            mockResponseHTTPCodeFromExternalClient(200);
+            Customer customer = customerClient.requestEligibilityCheck(request);
+            assertNotNull(customer);
+            assertEquals(true, customer.has("amount"));
+            assertEquals(true, customer.has("customer"));
+            String createRequest = getHost(Constants.ELIGIBILITY);
+            verifySentRequest(true, request.toString(), createRequest);
+        } catch (IOException e) {
+            assertTrue(false);
+        }
+    }
+
+    /**
+     * Fetch Eligibility
+     */
+    @Test
+    public void testFetchEligibility() throws RazorpayException {
+
+        String mockedResponseJson = "{\n" +
+                "  \"entity\": \"customer\",\n" +
+                "  \"amount\": \"500000\",\n" +
+                "  \"customer\": {\n" +
+                "    \"id\": \"KkBhM9EC1Y0HTm\",\n" +
+                "    \"contact\": \"+918220722114\"\n" +
+                "  },\n" +
+                "  \"instruments\": [\n" +
+                "    {\n" +
+                "      \"method\": \"emi\",\n" +
+                "      \"issuer\": \"HDFC\",\n" +
+                "      \"type\": \"debit\",\n" +
+                "      \"eligibility_req_id\": \"elig_KkCNLzlNeMYQyZ\",\n" +
+                "      \"eligibility\": {\n" +
+                "        \"status\": \"eligible\"\n" +
+                "      }\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+        try {
+            mockResponseFromExternalClient(mockedResponseJson);
+            mockResponseHTTPCodeFromExternalClient(200);
+            Customer fetch = customerClient.fetchEligibility(ELIGILITY_ID);
+            assertNotNull(fetch);
+            assertEquals(true, fetch.has("amount"));
+            assertEquals(true, fetch.has("customer"));
+            String fetchRequest = getHost(String.format(Constants.ELIGIBILITY_FETCH, ELIGILITY_ID));
+            verifySentRequest(false, null, fetchRequest);
```

Comment:
just want to understand what is the first parameter in verifySentRequest(), why sometimes it's true and sometimes it's false

Reply:
@rohitcbr When we are passing the request payload that time we have to pass true and if there is no payload then we pass false there

Reply:
got it, +1